docs/OV_Runtime_UG/performance_hints.md

```diff
@@ -0,0 +1,146 @@
+Every OpenVINO device supports these, which makes the things portable and future-proof.
+The also allows to do a performance configuration that is fully compatible with the [automatic device selection](./auto_device_selection.md).
+A special `ov::hint::PerformanceMode::UNDEFINED` acts same just as specifying no hint.
+
+Please also see the last section in the document on conducting the performance measurements with the `benchmark_app`.
+
+Notice that if there are other performance factors (other than inference time) like memory footprint and model load/compilation time are of concern, a typical model may take significantly more time to load with `ov::hint::PerformanceMode::THROUGHPUT` and then consume  much more memory, compared to the `ov::hint::PerformanceMode::LATENCY`.  
+
+## Performance Hints: How It Works?
+Internally, every device "translates" the value of the hint to the actual performance settings.
+For example the `ov::hint::PerformanceMode::THROUGHPUT` selects number of CPU or GPU streams.
+For the GPU, additionally the optimal batch size is selected and the [automatic batching](../OV_Runtime_UG/automatic_batching.md) is applied whenever possible (and also [when the device supports that](./supported_plugins/Device_Plugins.md#features-support-matrix)).
+
+The resulting (device-specific) settings can be queried back from the instance of the `ov:compiled_model`.  
+Notice that the `benchmark_app`, outputs the actual settings, for example:
+
+<code>
+$benchmark_app -hint tput -d CPU -m 'path to your favorite model'
+
+...
+
+[Step 8/11] Setting optimal runtime parameters
+
+[ INFO ] Device: CPU
+
+[ INFO ]   { PERFORMANCE_HINT , THROUGHPUT }
+
+...
+
+[ INFO ]   { OPTIMAL_NUMBER_OF_INFER_REQUESTS , 4 }
+
+[ INFO ]   { NUM_STREAMS , 4 }
+
+...
+</code> 
+
+## Using the Performance Hints: Basic API
+In the example code-snippet below the  `ov::hint::PerformanceMode::THROUGHPUT` is specified for the `ov::hint::performance_mode` property for the compile_model:
+@sphinxdirective
+
+.. tab:: C++
+
+    .. doxygensnippet:: docs/snippets/ov_auto_batching.cpp
+       :language: cpp
+       :fragment: [compile_model]
+
+.. tab:: Python
+
+    .. doxygensnippet:: docs/snippets/ov_auto_batching.py
+       :language: python
+       :fragment: [compile_model]
+
+@endsphinxdirective
+
+## Additional (Optional) Hints from the App
+Let's take an example  of an application that processes 4 video streams.  The most future-proof way to communicate the limitation of the parallel slack is to equip the performance hint with the optional `ov::hint::num_requests` configuration key set to 4. 
+As discussed previosly, for the GPU this will limit the batch size, for the CPU - the number of inference streams, so each device uses the `ov::hint::num_requests` while converting the hint to the actual device configuration options:
```

Comment:
```suggestion
For the GPU this will limit the batch size, for the CPU - the number of inference streams, so each device uses the `ov::hint::num_requests` while converting the hint to the actual device configuration options:
```